src/hotspot/os/aix/porting_aix.cpp

```diff
@@ -244,21 +235,16 @@ bool AixSymbols::get_function_name (
       }
       p_name[i] = '\0';
 
-      // If it is a C++ name, try and demangle it using the Demangle interface (see demangle.h).
-#ifndef DISABLE_DEMANGLE
+      // If it is a C++ name, try and demangle it using the __cxa_demangle interface(see demangle.h).
       if (demangle) {
-        char* rest;
-        Name* const name = Demangle(p_name, rest);
-        if (name) {
-          const char* const demangled_name = name->Text();
-          if (demangled_name) {
-            strncpy(p_name, demangled_name, namelen-1);
-            p_name[namelen-1] = '\0';
+	int status;
+	char *demangled_name = abi::__cxa_demangle(p_name, NULL, NULL, &status);
```

Comment:
AFAIK __cxa_demangle uses malloc/realloc. Below you are using delete, please do not mix them. Instead use ALLOW_C_FUNCTION(::free, ::free(demangled_name)).

Reply:
My mistake! I commented on an earlier version of the code in which Deepa used free, and asked them to change to delete. So, it is my misunderstanding here.

Thanks for clarifying that it depends on the allocator used rather than a feature of the project as I had thought.

Reply:
Thanks @jcking and @backwaterred for review the code.